satellite-software/src/LSM9DS1Sensor.cpp

```diff
@@ -349,4 +349,46 @@ void LSM9DS1Sensor::setGysroSensitivitySetting(GyroSensitivitySetting setting) {
 void LSM9DS1Sensor::setMagneticSensitivitySetting(MagneticSensitivitySetting setting) {
     eeprom_update_byte((uint8_t *)(getEEPROMBaseAddress()+OFFSET_MAGNETIC_SENSITIVITY), setting);
     _magneticSensitivity = setting;
+}
+
+uint8_t LSM9DS1Sensor::handleCommand(uint16_t cmdSequenceID, uint8_t command, uint8_t* recievedData, uint8_t recievedDataLen){
+    if(recievedDataLen != 1){
+        return CMD_STATUS_BAD_PARAM;
+    }
+    PRINT_DEBUG(F("Change LSM9DS1 settings"));
+    PRINT_DEBUG(F("\n"));
+    
+    switch(command){
+        case 0x00:
+        if(*recievedData > 0x04) return CMD_STATUS_BAD_PARAM;
+        PRINT_DEBUG(F("  Accel setting = "));
+        PRINT_DEBUG(*recievedData);
+        PRINT_DEBUG(F("\n"));
+        setAcceleratonSensitivitySetting((AccelerationSensitivitySetting) *recievedData);        
+        break;
+        case 0x01:
+        if(*recievedData != 0x00 && *recievedData != 0x10 && *recievedData != 0x20 && *recievedData != 0x40) return CMD_STATUS_BAD_PARAM;
```

Comment:
This is not the best way to accomplish what you are trying to do, as it will generate a lot of code for each distinct comparison. Use at bit mask here. 
```
if ( ((*recievedData)&0x8F != 0 ) {
    return CMD_STATUS_BAD_PARAM;
}
```
Basically, you are checking to see if any of the bits that allowed to be set are set. Granted, it doesn't check for allowed combinations of the allowed bits, but this is nowise than the checks you make for the other cases here where you effectively do the same thing as my suggestion but with a `>` check.

Reply:
Agree, might leave some gap but performance is more important.
Change made